packages/next/shared/lib/router/router.ts

```diff
@@ -1589,6 +1657,92 @@ export default class Router implements BaseRouter {
     }
   }
 
+  private async matchesMiddleware(asPath: string, locale?: string) {
+    const { pathname: asPathname } = parsePath(asPath)
+    const cleanedAs = removeLocale(
+      hasBasePath(asPathname) ? removeBasePath(asPathname) : asPathname,
+      locale
+    )
+
+    const fns = await this.pageLoader.getMiddlewareList()
```

Comment:
Could we remove the `getMiddlewareList` method and inline the regexes that should be matched against via an env variable like `__NEXT_MIDDLEWARE_REGEXES`?

This could help reduce the size a bit and remove an extra manifest needing to be loaded

Reply:
I think right now it works exactly as `pages` and for production it uses a predefined list of middleware.